scripts/main.py

```diff
@@ -46,15 +49,74 @@ def run(config: dict, cache: str):
     incident_model = LinearIncidentUptakeModel().fit(
         incident_train_data, grouping_factors
     )
-    cumulative_projections = incident_model.predict(
-        config["timeframe"]["start"],
-        config["timeframe"]["end"],
-        config["timeframe"]["interval"],
-        grouping_factors,
-    )
-    print(cumulative_projections)
-    incident_projections = cumulative_projections.to_incident(grouping_factors)
-    print(incident_projections)
+
+    option = config["option"]
+
+    if option == "projection":
+        cumulative_projections = incident_model.predict(
+            config["timeframe"]["start"],
+            config["timeframe"]["end"],
+            config["timeframe"]["interval"],
+            grouping_factors,
+        )
+        print(cumulative_projections)
+        incident_projections = cumulative_projections.to_incident(grouping_factors)
+        print(incident_projections)
+
+    elif option == "evaluation":
+        # Make projections sequentially
+        interval_str = config["timeframe"]["interval"]
+        interval_number = int(interval_str[0])
+
+        dates = pl.date_range(
+            config["timeframe"]["start"],
+            config["timeframe"]["end"] - dt.timedelta(interval_number),
+            config["timeframe"]["interval"],
+            eager=True,
+        )
+
+        scores = pl.DataFrame(
+            schema={
+                "forecast_start": pl.Date,
+                "forecast_end": pl.Date,
+                "score": pl.Float64,
+                "type": pl.String,
+            }
+        )
+
+        for date in dates:
```

Comment:
I strongly recommend you move away from for loops and toward functions here

This is where the Makefile or Snakemake will come in handy. Like, you're doing many different jobs:

- For each combination of forecast date and model, do the forecast
- For each model and score function (and forecast horizon?), score the model using the forecasts from the previous step
- Report these out in a table?
- Make some figures?

So rather than saying "what do I need to do for each date?" and putting that into a for loop, instead say "what are the different jobs I need to do?" and write a function for each one.

There will be looping logic somewhere, but it might be in the Makefile or snakefile

Reply:
Why is it better to have looping logic in a Makefile than in the script? Just so that iterations of the loop that have already been completed aren't repeated?

Reply:
Does for loop play as a high-level command so it should be in the Makefile, while low-level command should be functions simple enough to debug?

Reply:
New issue created https://github.com/CDCgov/cfa-immunization-uptake-projection/issues/86#issue-2748546812

Reply:
I agree it's fine to postpone the relocation of looping for the next PR, because it could be influenced by our redistribution of `main.py` into 3-4 separate scripts.